### PR TITLE
[dnf5] Implement new argument "--dump-variables"

### PR DESCRIPTION
**Pull request by jrohel**
If dnf5 is run with this argument, variable values are added to standard output.

The old DNF4 supports this argument only with the `config-manager` plugin. If the user wants to know the values of the variables with which a certain command will run, he must first use `config-manager` and then run the desired command with the same settings (same environment and configuration files) and hope that nothing changed in between (e.g. thanks to a plugin).

In other words, this PR implements part of the functionality of the dnf4 `config-manager` plugin, but for general use.

Related to: https://github.com/rpm-software-management/dnf5/issues/405

### CODE REVIEW

**Comment by m-blaha** [comment]
Just an idea - would it make sense to implement this as a new command (e.g. `dnf5 debug`)? It could present to the user more information - variables, all applied config options (since configuration is spread among multiple files, this might be interesting), versions of libraries in use (libdnf5, libsolv, librpm, librepo...) and so on.

Copr plugin has the same subcommand used for similar purpose.

**Comment by jrohel** [comment]
Rebased to resolve conflicts.

**Comment by ppisar** [comment]
There is missing a documentation in doc/dnf5.8.rst.

**Comment by ppisar** [comment]
I somehow find it quite obtrusive that "./dnf5/dnf5 --dump-variables" does not work also alone:

~~~~
$ ./dnf5/dnf5 --dump-variables
Missing command. Add "--help" for more information about the arguments.
~~~~

But maybe that's on purpose.

**Comment by jrohel** [comment]
@ppisar
I intended that `--dump-variables` would print out what variables the given command was running with.
I am considering whether to modify the behavior of dnf5. Now a command must always be entered, otherwise dnf5 will print "Missing command..."

**Review by ppisar — APPROVED** [review]
Documentation and standalone invocation works now.
